application.py

```diff
@@ -305,7 +312,24 @@ def upload_strategy():
     (pylint_stdout, _) = lint.py_run(local_path, return_std=True)
     pylint_message = pylint_stdout.read()
 
-    strategy_folder = os.path.join(userid, new_folder)
+    test_id = request.args.get('test_id')
+    if test_id is not None:
+        test_id = int(test_id)
+        logger.info("uploading testing file...")
+        cnt_loc = test_id
+    else:
+        logger.info("uploading user file...")
+        cursor.execute(
+            "SELECT MAX(strategy_id) as m FROM backtest.strategies"
+        )
+        for id in cursor.fetchall():
+            cnt_loc = id['m']
```

Comment:
can we rename `m` to `max_strategy_id`

Reply:
use `fetchone` https://dev.mysql.com/doc/connector-python/en/connector-python-api-mysqlcursor-fetchone.html